tests/test_tokenization_fast.py

```diff
@@ -51,3 +56,37 @@ def test_prepare_for_model(self):
     def test_rust_tokenizer_signature(self):
         # PreTrainedTokenizerFast doesn't have tokenizer_file in its signature
         pass
+
+    def test_training_new_tokenizer(self):
+        tmpdirname_orig = self.tmpdirname
+        # Here we want to test the 2 available tokenizers that use 2 different types of models: Unigram and WordLevel.
+        for tokenizer, pretrained_name, kwargs in self.tokenizers_list:
+            with self.subTest(f"{tokenizer.__class__.__name__} ({pretrained_name})"):
+                try:
+                    self.tmpdirname = tempfile.mkdtemp()
+                    tokenizer = self.rust_tokenizer_class.from_pretrained(pretrained_name, **kwargs)
+
+                    tokenizer.save_pretrained(self.tmpdirname)
+                    super().test_training_new_tokenizer()
+                finally:
+                    # Even if the test fails, we must be sure that the folder is deleted and that the default tokenizer
+                    # is restored
+                    shutil.rmtree(self.tmpdirname)
+                    self.tmpdirname = tmpdirname_orig
+
+    def test_training_new_tokenizer_with_special_tokens_change(self):
+        tmpdirname_orig = self.tmpdirname
+        # Here we want to test the 2 available tokenizers that use 2 different types of models: Unigram and WordLevel.
+        for tokenizer, pretrained_name, kwargs in self.tokenizers_list:
+            with self.subTest(f"{tokenizer.__class__.__name__} ({pretrained_name})"):
+                try:
+                    self.tmpdirname = tempfile.mkdtemp()
+                    tokenizer = self.rust_tokenizer_class.from_pretrained(pretrained_name, **kwargs)
+
+                    tokenizer.save_pretrained(self.tmpdirname)
+                    super().test_training_new_tokenizer_with_special_tokens_change()
+                finally:
+                    # Even if the test fails, we must be sure that the folder is deleted and that the default tokenizer
+                    # is restored
+                    shutil.rmtree(self.tmpdirname)
+                    self.tmpdirname = tmpdirname_orig
```

Comment:
I propose here to change the tests of `PreTrainedTokenizerFast` but we could also have changed directly the commons tests. I didn't do it because I have the impression that generally the models stored in `self.tmpdirname` are often toy examples, while in `self.tokenizers_list` larger models would be used. Besides, if we ever moved the change to the commons tests we would have to fix an error with the GPT2 tokenizer.

Reply:
I think it's good the way you did it.